File: cpp/unittests/HouseRobberTest.cpp
#include "HouseRobber.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(HouseRobber, Solution198_1) {
  std::vector<int> Nums1{1, 2, 3, 1};
  const int Expected1 = 4;
  EXPECT_EQ(Expected1, Solution198_1().rob(Nums1));

  std::vector<int> Nums2{2, 7, 9, 3, 1};
  const int Expected2 = 12;
  EXPECT_EQ(Expected2, Solution198_1().rob(Nums2));
}

File: java/src/test/java/io/binac/leetcode/HouseRobberTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class HouseRobberTest {
    private static final HouseRobber.Solution1 SOLUTION1 = new HouseRobber.Solution1();

    private final int[] NUMS1 = {1, 2, 3, 1};
    private final int EXPECTED1 = 4;

    private final int[] NUMS2 = {2, 7, 9, 3, 1};
    private final int EXPECTED2 = 12;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.rob(NUMS1));
        assertEquals(EXPECTED2, SOLUTION1.rob(NUMS2));
    }
}